src/components/Button/Button.styled.js

```diff
@@ -0,0 +1,85 @@
+import styled, { css } from "styled-components";
+import { capitalize } from "./../../utils/utils";
+
+const themeToCSS = {
+  variant: {
+    "--background-color": "buttonBackgroundColor",
+    "--border-color": "buttonBorderColor",
+    "--hover-background-color": "buttonHoverBackgroundColor",
+    "--hover-border-color": "buttonHoverBorderColor",
+  },
+  size: {
+    "--padding": "buttonPadding",
+    "--font-size": "buttonFontSize",
+    "--border-radius": "buttonBorderRadius",
+  },
+};
+
+const generateStyles = props => {
+  let styles = "";
+
+  if (props.theme) {
+    for (const propsName in themeToCSS) {
+      if (props[propsName]) {
+        for (const variable in themeToCSS[propsName]) {
+          let themeKey =
+            themeToCSS[propsName][variable] + capitalize(props[propsName]);
+
+          if (propsName === "variant") {
+            if (props.active) {
+              themeKey += "Active";
+            } else if (props.disabled) {
+              themeKey += "Disabled";
+            }
+          }
+
+          styles += `${variable}: ${props.theme[themeKey]};`;
+        }
+      }
+    }
+
+    if (props.disabled) {
+      styles += `
+        opacity: .65;
+      `;
+    }
+  }
+
+  return css`
+    ${styles}
+  `;
+};
```

Comment:
Bartku, skoro to rozwiązania działa to przygotowałbym sobie jakiś "util", który będę mógł wykorzystać wielokrotnie.
Możesz je też wykorzystać wtedy w `Alert`. Postaraj się uprościć kod i rozdzielić go na mniejsze części.
Ilość ifów jest tam porażająca (tak pamiętam, że to część mojego kodu :D)